test/kitchen/kitchen-azure-chef-test.yml

```diff
@@ -4,7 +4,9 @@ suites:
 - name: dd-agent
   run_list:
     - "recipe[dd-agent-sles-workaround]"
+    - "recipe[dd-agent-system-files-check::list-files-before-install]"
     - "recipe[dd-agent-install]"
+    - "recipe[dd-agent-system-files-check::check-files-after-uninstall]"
```

Comment:
I believe the Agent is still installed at this point of the Chef run: the uninstall is done in the tests (otherwise we wouldn't be able to test anything about the installed Agent). See: https://github.com/DataDog/datadog-agent/blob/70bebc3d709cc79a38876c6a43b81e4a3a1a0416/test/kitchen/test/integration/common/rspec/spec_helper.rb#L642-L659.

It would be nice to change that though (maybe split in 2 test suites, with one dedicated to the uninstall, but that would increase resource usage).

Reply:
Good catch, I will change that as part of this PR.

Reply:
Actually I won't, kitchen is a lost cause.